### PR TITLE
fix(components): export foundation element definition

### PR DESCRIPTION
**Pull request by yinonov**
- [x] add tests to ensure elements definitions and design system reference are exported

### CODE REVIEW

File: change/@vonage-vivid-c57b0cba-e40d-4f8c-984c-b885be0c7b6c.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "rewrite component definition",
```
Comment (yinonov) [inline change/@vonage-vivid-c57b0cba-e40d-4f8c-984c-b885be0c7b6c.json:3]:
```suggestion
  "comment": "export component definition and design system",
```

File: change/@vonage-nx-vivid-987f3d5d-1355-4160-82c2-e29335e54634.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "comment": "align with public exports",
```
Comment (yinonov) [inline change/@vonage-nx-vivid-987f3d5d-1355-4160-82c2-e29335e54634.json:2]:
```suggestion
  "comment": "rewrite component definition",
```

File: libs/components/src/lib/number-field/definition.ts [hunk 3]
```diff
@@ -1,32 +1,30 @@
...
 		template: template as any,
 		styles,
 		shadowOptions: {
 			delegatesFocus: true,
 		},
-	})();
+	});
 
-export const numberFieldElement = [numberField, ...buttonElements, ...dividerElements];
+export const numberFieldElement = [numberFieldDefinition(), ...buttonRegistries, ...dividerRegistries];
```
Comment (olaf-k) [inline libs/components/src/lib/number-field/definition.ts:23]:
```suggestion
export const numberFieldRegistries = [numberFieldDefinition(), ...buttonRegistries, ...dividerRegistries];
```

File: libs/components/src/lib/number-field/number-field.spec.ts [hunk 4]
```diff
@@ -20,14 +22,23 @@ function getRootElement(element: NumberField) {
 
 describe('vwc-number-field', () => {
 
+	/**
+	 *
+	 */
```
Comment (olaf-k) [inline libs/components/src/lib/number-field/number-field.spec.ts:27]: to be removed

File: libs/components/src/lib/number-field/number-field.spec.ts [hunk 5]
```diff
@@ -20,14 +22,23 @@ function getRootElement(element: NumberField) {
...
 	function setToBlurred() {
 		element.dispatchEvent(new Event('blur'));
 	}
 
+	/**
+	 *
+	 */
```
Comment (olaf-k) [inline libs/components/src/lib/number-field/number-field.spec.ts:34]: to be removed

File: libs/components/src/lib/number-field/number-field.spec.ts [hunk 6]
```diff
@@ -560,6 +572,9 @@ describe('vwc-number-field', () => {
 		});
 
 		it('should set inert in disabled and readonly', async function() {
+			/**
+			 *
+			 */
```
Comment (olaf-k) [inline libs/components/src/lib/number-field/number-field.spec.ts:577]: to be removed

File: libs/components/src/lib/number-field/number-field.spec.ts [hunk 7]
```diff
@@ -582,6 +597,9 @@ describe('vwc-number-field', () => {
 		});
 
 		it('should set tabindex="-1" on the buttons', async function() {
+			/**
+			 *
+			 */
```
Comment (olaf-k) [inline libs/components/src/lib/number-field/number-field.spec.ts:602]: to be removed

File: libs/components/src/shared/design-system/index.spec.ts [hunk 8]
```diff
@@ -1,11 +1,15 @@
...
 
 describe('design system', () => {
+	it('should export \'designSystem\'', async () => {
+		expect(designSystem).toBeDefined();
+	});
+
 	it('should register with custom prefix instead of default', async () => {
-		const elementsDefintions = [badge, icon];
-		const registerBadge = registerFactorial(elementsDefintions);
+		const elementsDefintions = [badgeDefinition(), iconDefinition()];
```
Comment (olaf-k) [inline libs/components/src/shared/design-system/index.spec.ts:11]: all over this file: `s/elementsDefintions/elementsDefinitions`

File: libs/nx-vivid/src/generators/component/files/__fileName__.spec.ts__tmpl__ [hunk 9]
```diff
@@ -13,7 +13,8 @@ describe( 'vwc-<%= name %>', () => {
 
 	describe('basic', () => {
 		it('should be initialized as a vwc-<%= name %>', async () => {
-			expect(element).toBeInstanceOf(<%= className %>);
+			expect((<%= className %>)Definition()).toBeInstanceOf(FoundationElementRegistry);
```
Comment (olaf-k) [inline libs/nx-vivid/src/generators/component/files/__fileName__.spec.ts__tmpl__:16]: needs `import { FoundationElementRegistry } from '@microsoft/fast-foundation';` then

**Review by olaf-k — CHANGES_REQUESTED** [review]

File: libs/components/src/shared/design-system/index.spec.ts (hunk 8 above)
Comment (yinonov) [reply libs/components/src/shared/design-system/index.spec.ts:11]: what?

File: libs/components/src/lib/number-field/number-field.spec.ts (hunk 6 above)
Comment (yinonov) [reply libs/components/src/lib/number-field/number-field.spec.ts:571]:
```suggestion
```

File: libs/components/src/shared/design-system/index.spec.ts (hunk 8 above)
Comment (olaf-k) [reply libs/components/src/shared/design-system/index.spec.ts:11]: small typo: `elementsDefintions` should be searched/replaced by `elementsDefinitions` (note the additional `i`)

**Review by olaf-k — APPROVED** [review]

**Pull request merged**